apps/wallet/src/ui/app/pages/home/transfer-coin/SendTokenForm.tsx

```diff
@@ -0,0 +1,334 @@
+    gasInputBudgetEst: null as number | null,
+};
+
+export type FormValues = typeof initialValues;
+
+export type SubmitProps = {
+    to: string;
+    amount: string;
+    isPayAllSui: boolean;
+    coinIds: string[];
+    gasBudget: number;
+};
+
+export type SendTokenFormProps = {
+    coinType: string;
+    onSubmit: (values: SubmitProps) => void;
+    initialAmount: string;
+    initialTo: string;
+};
+
+function GasBudgetEstimation({
+    coinDecimals,
+    suiCoins,
+}: {
+    coinDecimals: number;
+    suiCoins: SuiMoveObject[];
+}) {
+    const { values, setFieldValue } = useFormikContext<FormValues>();
+
+    const gasBudgetEstimationUnits = useMemo(
+        () =>
+            Coin.computeGasBudgetForPay(
+                suiCoins,
+                parseAmount(values.amount, coinDecimals)
+            ),
+        [coinDecimals, suiCoins, values.amount]
+    );
+    const { gasBudget: gasBudgetEstimation, isLoading } = useGasBudgetInMist(
+        gasBudgetEstimationUnits
+    );
+
+    const [formattedGas, gasSymbol] = useFormatCoin(
+        gasBudgetEstimation,
+        GAS_TYPE_ARG
+    );
+
+    // gasBudgetEstimation should change when the amount above changes
+    useEffect(() => {
+        setFieldValue('gasInputBudgetEst', gasBudgetEstimation);
+    }, [gasBudgetEstimation, setFieldValue, values.amount]);
+
+    return (
+        <Loading loading={isLoading}>
+            <div className="px-2 mt-3 mb-5 flex w-full gap-2 justify-between">
+                <div className="flex gap-1">
+                    <Text variant="body" color="gray-80" weight="medium">
+                        Estimated Gas Fees
+                    </Text>
+                    <div className="text-gray-60 h-4 items-end flex">
+                        <IconTooltip tip="Estimated Gas Fees" placement="top" />
+                    </div>
+                </div>
+                <Text variant="body" color="gray-90" weight="medium">
+                    {formattedGas} {gasSymbol}
+                </Text>
+            </div>
+        </Loading>
+    );
+}
+
+// Set the initial gasEstimation from initial amount
+// base on the input amount feild update the gasEstimation value
+// Seperating the gasEstimation from the formik context to access the input amount value and update the gasEstimation value
+export function SendTokenForm({
+    coinType,
+    onSubmit,
+    initialAmount = '',
+    initialTo = '',
+}: SendTokenFormProps) {
+    const aggregateBalances = useAppSelector(accountAggregateBalancesSelector);
+    const coinBalance = useMemo(
+        () => (coinType && aggregateBalances[coinType]) || BigInt(0),
+        [coinType, aggregateBalances]
+    );
+
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const coins = useMemo(
+        () =>
+            allCoins.filter(
+                ({ type }) => CoinAPI.getCoinType(type) === coinType
+            ),
+        [allCoins, coinType]
+    );
+
+    const suiCoins = useMemo(
```

Comment:
I think we can remove this. For the case we use it coinType will be sui and coins and suiCoins will be the same array